Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/connected_components.h

```diff
@@ -956,6 +962,180 @@ void keep_connected_components(PolygonMesh& pmesh
+    CGAL::copy_face_graph(filter_graph, new_graph);
+  }
+}
+}//internal
+
+/*!
+ * \ingroup keep_connected_components_grp
+ * identifies the connected components of `pm` and pushes back a new `PolygonMesh` for each connected component in `cc_meshes`.
+ *
+ *
+ *  \tparam PolygonMesh a model of `FaceListGraph`
+ *  \tparam PolygonMeshRange a model of `SequenceContainer` (providing
+ *  `void push_back()` and `PolygonMesh& back()`) with `PolygonMesh` as value type.
+ *
+ *  \tparam NamedParameters a sequence of Named Parameters
+ *
+ * \param pm the polygon mesh
+ * \param cc_meshes container that is filled with the extracted connected components.
+ * \param np an optional sequence of Named Parameters among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *   \cgalParamBegin{edge_is_constrained_map} a property map containing the constrained-or-not status of each edge of `pm` \cgalParamEnd
+ *   \cgalParamBegin{face_index_map}
+ *     a property map containing an index for each face initialized from 0 to `num_faces(pm)`
+ *   \cgalParamEnd
+ *   \cgalParamBegin{vertex_index_map}
+ *     a property map containing an index for each vertex initialized 0 to `num_vertices(pm)`
+ *   \cgalParamEnd
+ *   \cgalNPBegin{halfedge_index_map}
+ *     a property map containing an index for each halfedge initialized 0 to `num_halfedges(pm)`
+ *   \cgalNPEnd
+ * \cgalNamedParamsEnd
+ *
+ */
+template <class PolygonMesh, class PolygonMeshRange, class NamedParameters>
+void split_connected_components(const PolygonMesh& pm,
+                                PolygonMeshRange& cc_meshes,
+                                const NamedParameters& np)
+{
+  typedef typename boost::mpl::if_c<CGAL::graph_has_property<PolygonMesh, CGAL::face_index_t>::value
+        , CGAL::face_index_t
+        , CGAL::dynamic_face_property_t<std::size_t >
+        >::type FIM_def_tag;
+
+  typedef typename boost::mpl::if_c<CGAL::graph_has_property<PolygonMesh, CGAL::halfedge_index_t>::value
+        , CGAL::halfedge_index_t
+        , CGAL::dynamic_halfedge_property_t<std::size_t >
+        >::type HIM_def_tag;
+
+  typedef typename boost::mpl::if_c<CGAL::graph_has_property<PolygonMesh, boost::vertex_index_t>::value
+        , boost::vertex_index_t
+        , CGAL::dynamic_vertex_property_t<std::size_t >
+        >::type VIM_def_tag;
+
+  typedef typename internal_np::Lookup_named_param_def <
+    internal_np::edge_is_constrained_t,
+    NamedParameters,
+    internal::No_mark<PolygonMesh>//default
+  > ::type Ecm;
+
+  Ecm ecm = parameters::choose_parameter( parameters::get_parameter(np, internal_np::edge_is_constrained),
+                                   internal::No_mark<PolygonMesh>() );
+
+  internal::split_connected_components_impl(
+        internal::get_index_map(
+          parameters::get_parameter(np, internal_np::face_index),
+          FIM_def_tag(),
+          CGAL::dynamic_face_property_t<std::size_t >(),
+          pm, faces(pm)),
```

Comment:
I'm in favor of a small moratorium on these new maps/helper functions as I think we are venturing into super-unreadable territory and I think it needs a better design.

Reply:
Proper API will be available once https://github.com/CGAL/cgal/pull/4209 is integrated.